apps/server/src/controllers/integrationController.ts

```diff
@@ -222,6 +237,51 @@ export function dispatchFromAdapter(type: string, payload: unknown, source?: 'os
       return { topic: 'timer', payload: timer };
     }
 
+    // ontime/change/{eventID}/{propertyName}
+    case 'change': {
+      try {
+        if (params.length < 2) {
+          throw new Error('Invalid parameters');
+        }
+
+        const eventID = params[0];
+        const propertyName = params[1] as keyof OntimeEvent;
+
+        const payloadMap = Object.fromEntries(
+          Object.entries(event).map(([key, value]) => {
+            return [key, typeof value];
+          }),
+        );
+
+        if (!payloadMap.hasOwnProperty(propertyName)) {
+          throw new Error(`Cannot update unknown event property ${propertyName}`);
+        }
```

Comment:
thinking out loud, could we make a type guard here `isKeyOfOntimeEvent`, this would give us the validation and the type at once and replace all this logic